cirq/google/calibration/phased_fsim.py

```diff
@@ -76,3 +85,102 @@ def override_by(self, other: 'PhasedFSimParameters') -> 'PhasedFSimParameters':
             Otherwise the current values are used.
         """
         return other.merge_with(self)
+
+
+@json_serializable_dataclass(frozen=True)
+class FloquetPhasedFSimCalibrationOptions:
+    characterize_theta: bool
+    characterize_zeta: bool
+    characterize_chi: bool
+    characterize_gamma: bool
+    characterize_phi: bool
+
+
+# TODO: Fix json serialization (the default one doesn't work with tuples as dictionary keys).
+# TODO: Add start and end calibration timestamp
+# TODO: Add export to Panda's data frame
+@json_serializable_dataclass(frozen=True)
+class PhasedFSimCalibrationResult:
+    # TODO: Instead of parameters and gate, should that just be cirq.Moment validated at construction time? (This is
+    #  slightly problematic with frozen data classes).
+    parameters: Dict[Tuple[Qid, Qid], PhasedFSimParameters]
+    gate: Gate
+    gate_set: SerializableGateSet
+
+    def get_parameters(self, a: Qid, b: Qid) -> Optional['PhasedFSimParameters']:
+        if (a, b) in self.parameters:
+            return self.parameters[(a, b)]
+        elif (b, a) in self.parameters:
+            return self.parameters[(b, a)].parameters_for_qubits_swapped()
+        else:
+            return None
+
+
+@json_serializable_dataclass(frozen=True)
+class PhasedFSimCalibrationRequest(abc.ABC):
+    gate: Gate  # Any gate which can be described by cirq.PhasedFSim
+    gate_set: SerializableGateSet
+    pairs: Tuple[Tuple[Qid, Qid]]
+
+    @abc.abstractmethod
+    def to_calibration_layer(self, handler_name: str) -> CalibrationLayer:
+        pass
+
+    @abc.abstractmethod
+    def parse_result(self, result: CalibrationResult) -> PhasedFSimCalibrationResult:
+        pass
+
+
+@json_serializable_dataclass(frozen=True)
+class FloquetPhasedFSimCalibrationResult(PhasedFSimCalibrationResult):
+    options: FloquetPhasedFSimCalibrationOptions
+
+
+@json_serializable_dataclass(frozen=True)
+class FloquetPhasedFSimCalibrationRequest(PhasedFSimCalibrationRequest):
+    options: FloquetPhasedFSimCalibrationOptions
+
+    def to_calibration_layer(self, handler_name: str) -> CalibrationLayer:
+        circuit = Circuit([self.gate.on(*pair) for pair in self.pairs])
+        return CalibrationLayer(
+            calibration_type='floquet_phased_fsim_characterization',
```

Comment:
Should we make 'floquet_phased_fsim_characterization' a constant somewhere?  It looks like we will need it for  run_characterizations as well.

Reply:
It was my oversight to put this string here, thanks for pointing out. My plan is to remove "handler_name" argument and make it a constant as you're suggesting.

Reply:
Done.